documentation/specs/src/interoperability/ethereum-bridge/transfers_to_namada.md

```diff
@@ -0,0 +1,58 @@
+# Transferring assets from Ethereum to Namada
+
+In order to facilitate transferring assets from Ethereum to Namada, There 
+ will be two internal accounts with associated native validity predicates:
+
+- `#EthBridge` - Controls the `/eth_msgs/` [storage](ethereum_events_attestation.md/#storage) 
+- and ledgers of balances
+  for wrapped Ethereum assets (ERC20 tokens) structured in a
+  ["multitoken"](https://github.com/anoma/anoma/issues/1102) hierarchy
+- `#EthBridgeEscrow` which will hold in escrow wrapped Namada tokens which have
+  been sent to Ethereum.
+
+#### Wrapped ERC20
+
+If an ERC20 token is transferred to Namada, once the associated 
+`TransferToNamada` Ethereum event is included into Namada, validators mint 
+the  appropriate amount to the corresponding  multitoken balance key for 
+the receiver.
+
+```rust
+pub struct EthAddress(pub [u8; 20]);
+
+/// Represents Ethereum assets on the Ethereum blockchain
+pub enum EthereumAsset {
+    /// An ERC20 token and the address of its contract
+    ERC20(EthAddress),
+}
+
+/// An event transferring some kind of value from Ethereum to Anoma
+pub struct TransferToNamada {
+    /// Quantity of ether in the transfer
+    pub amount: Amount,
+    /// Address on Ethereum of the asset
+    pub asset: EthereumAsset,
+    /// The Namada address receiving wrapped assets on Anoma
+    pub receiver: Address,
+}
```

Comment:
```suggestion
/// An event transferring some kind of value from Anoma to Ethereum
pub struct TransferToEthereum {
    /// Quantity of wrapped Asset in the transfer
    pub amount: Amount,
    /// Address of the smart contract issuing the token
    pub asset: EthAddress,
    /// The address receiving assets on Ethereum
    pub receiver: EthAddress,
}
```
to update: `asset` will always be an ERC20 address (at least for this initial version of the bridge). In the case of transferring NAM to Ethereum, `asset` is the ERC20 address of wNAM which is set when we deploy the Ethereum bridge smart contracts.

https://github.com/anoma/ethereum-bridge/blob/a1fcd58f5f51f161d9d70a142aa0e9bc0ae5a52e/scripts/deploy.js#L139-L140

Reply:
This should be a transfer to Namada, not Ethereum.

Reply:
sorry the struct should be
```rust
pub struct TransferToNamada {
    /// Quantity of the ERC20 token in the transfer
    pub amount: Amount,
    /// Address of the smart contract issuing the token
    pub asset: EthAddress,
    /// The address receiving wrapped assets on Anoma
    pub receiver: Address,
}
```